expertsystem/data.py

```diff
@@ -111,6 +112,43 @@ def __hash__(self) -> int:
         return hash(repr(self))
 
 
+@dataclass(frozen=True, init=False)
+class EdgeQuantumNumbers:  # pylint: disable=too-many-instance-attributes
+    """Definition of quantum numbers for edges."""
+
+    pid = NewType("pid", int)
+    energy = NewType("energy", complex)
+    mass = NewType("mass", float)
+    width = NewType("width", float)
+    spin_magnitude = NewType("spin_magnitude", float)
+    spin_projection = NewType("spin_projection", float)
+    charge = NewType("charge", int)
+    isospin_magnitude = NewType("isospin_magnitude", float)
+    isospin_projection = NewType("isospin_projection", float)
+    strangeness = NewType("strangeness", int)
+    charmness = NewType("charmness", int)
+    bottomness = NewType("bottomness", int)
+    topness = NewType("topness", int)
+    baryon_number = NewType("baryon_number", int)
+    electron_lepton_number = NewType("electron_lepton_number", int)
+    muon_lepton_number = NewType("muon_lepton_number", int)
+    tau_lepton_number = NewType("tau_lepton_number", int)
+    parity = NewType("parity", Parity)
+    c_parity = NewType("c_parity", Parity)
+    g_parity = NewType("g_parity", Parity)
+
+
+@dataclass(frozen=True, init=False)
+class NodeQuantumNumbers:
+    """Definition of quantum numbers for interaction nodes."""
+
+    l_magnitude = NewType("l_magnitude", float)
+    l_projection = NewType("l_projection", float)
+    s_magnitude = NewType("s_magnitude", float)
+    s_projection = NewType("s_projection", float)
+    parity_prefactor = NewType("parity_prefactor", float)
```

Comment:
I'm not sure what this comment is referring to?

Reply:
Ah the energy of the `EdgeQuantumNumbers`

Reply:
That is indeed a left over part from some experimentation, I will remove it

Reply:
4965f5447a8454c2ba5fb0d425998de01472bd21

Reply:
Actually my question was more about
```python
    @property
    def energy(self) -> complex:
        return complex(self.mass, self.width)
```
but somehow the lines got messed up.

I'll remove it in a PR to your fork.

Reply:
The idea behind it was supposed to go into the direction of a pole in the complex energy plane

Reply:
but `Particle` is interface to the outside, it's not used in the internal implementation

Reply:
By interface to the outside I (vaguely) mean: 'how would the PDG define this particle?' I don't think it will mention "the complex energy of particle X" but rather mass and width.